#! /bin/sh /usr/share/dpatch/dpatch-run
## 20_gcc43-missing-include.dpatch by Krzysztof Burghardt <krzysztof@burghardt.pl>
##
## DP: Fixes problems with compilation using GCC 4.3

@DPATCH@
diff -urNad poco-1.3.3p1~/Foundation/include/Poco/BinaryWriter.h poco-1.3.3p1/Foundation/include/Poco/BinaryWriter.h
--- poco-1.3.3p1~/Foundation/include/Poco/BinaryWriter.h	2008-10-09 18:14:00.000000000 +0200
+++ poco-1.3.3p1/Foundation/include/Poco/BinaryWriter.h	2008-11-15 11:28:14.487089108 +0100
@@ -42,7 +42,7 @@
 
 #include "Poco/Foundation.h"
 #include <ostream>
-
+#include <cstring>
 
 namespace Poco {
 
diff -urNad poco-1.3.3p1~/Foundation/include/Poco/File.h poco-1.3.3p1/Foundation/include/Poco/File.h
--- poco-1.3.3p1~/Foundation/include/Poco/File.h	2008-10-09 18:14:00.000000000 +0200
+++ poco-1.3.3p1/Foundation/include/Poco/File.h	2008-11-15 11:28:14.487089108 +0100
@@ -43,6 +43,7 @@
 #include "Poco/Foundation.h"
 #include "Poco/Timestamp.h"
 #include <vector>
+#include <cstring>
 
 
 #if defined(POCO_OS_FAMILY_WINDOWS) && defined(POCO_WIN32_UTF8)
diff -urNad poco-1.3.3p1~/Foundation/src/Environment.cpp poco-1.3.3p1/Foundation/src/Environment.cpp
--- poco-1.3.3p1~/Foundation/src/Environment.cpp	2008-10-09 18:14:00.000000000 +0200
+++ poco-1.3.3p1/Foundation/src/Environment.cpp	2008-11-15 11:28:42.199079948 +0100
@@ -35,6 +35,7 @@
 
 
 #include "Poco/Environment.h"
+#include <cstdio>
 #include <cstdlib>
 
 
diff -urNad poco-1.3.3p1~/XML/include/Poco/DOM/ElementsByTagNameList.h poco-1.3.3p1/XML/include/Poco/DOM/ElementsByTagNameList.h
--- poco-1.3.3p1~/XML/include/Poco/DOM/ElementsByTagNameList.h	2008-10-09 18:14:01.000000000 +0200
+++ poco-1.3.3p1/XML/include/Poco/DOM/ElementsByTagNameList.h	2008-11-15 11:28:14.487089108 +0100
@@ -44,6 +44,7 @@
 #include "Poco/DOM/NodeList.h"
 #include "Poco/XML/XMLString.h"
 
+#include <climits>
 
 namespace Poco {
 namespace XML {
